Lib9c/Action/Garages/LoadIntoMyGarages.cs

```diff
@@ -247,40 +247,33 @@ public override IAccount Execute(IActionContext context)
                 FungibleIdAndCounts);
             foreach (var (fungibleId, count, garageAddr, garageState) in fungibleItemTuples)
             {
-                if (!inventory.TryGetTradableFungibleItems(
-                        fungibleId,
-                        requiredBlockIndex: null,
-                        blockIndex: blockIndex,
-                        out var outItems))
+                if (!inventory.TryGetFungibleItems(fungibleId, out var outItems))
                 {
                     throw new ItemNotFoundException(InventoryAddr.Value, fungibleId);
                 }
 
-                var itemArr = outItems as Inventory.Item[] ?? outItems.ToArray();
-                var tradableFungibleItem = (ITradableFungibleItem)itemArr[0].item;
-                if (!inventory.RemoveTradableFungibleItem(
+                if (!inventory.RemoveFungibleItem(
                         fungibleId,
-                        requiredBlockIndex: null,
                         blockIndex: blockIndex,
                         count))
                 {
                     throw new NotEnoughItemException(
                         InventoryAddr.Value,
                         fungibleId,
                         count,
-                        itemArr.Sum(item => item.count));
+                        outItems.Sum(i => i.count));
                 }
 
-                IFungibleItem fungibleItem = tradableFungibleItem switch
+                var item = outItems[0].item;
+                if (item is not Material material)
                 {
-                    TradableMaterial tradableMaterial => new Material(tradableMaterial),
-                    _ => throw new InvalidCastException(
-                        $"Invalid type of {nameof(tradableFungibleItem)}: " +
-                        $"{tradableFungibleItem.GetType()}")
-                };
+                    throw new InvalidCastException(
+                        $"Invalid type of {nameof(item)}: " +
+                        $"{item.GetType()}");
+                }
```

Comment:
`Inventory.TryGetFungibleItems()`로 얻는 `List<Inventory.Item> 객체`는 `Material`과 `TradableMaterial`을 모두 포함하고 있어요. 그렇기 때문에 `0번 Inventory.Item.Item`이 `TradableMaterial`일 경우에는 이를 `Material로 바꿔서` `FungibleItemGarage` 생성자의 인자로 사용해야 해요.
//
The `List<Inventory.Item> object` we get from `Inventory.TryGetFungibleItems()` contains both `Material` and `TradableMaterial`, so if `Inventory.Item.Item` at number 0 is `TradableMaterial`, we need to replace it with `Material` and use it as an argument to the `FungibleItemGarage` constructor.

Reply:
TradableMaterial을 그대로 개러지에 올리면 문제가 발생하나요?

Reply:
개러지에 올라간 아이템을 이동(unload로 내 인벤토리로 되돌릴 경우, deliver로 다른 개러지로 전송할 경우)할때는 아이템의 속성이 변하지 않는걸로 이해했습니다. 이 코멘트를 반영하면 TradableMaterial을 올리고 되돌려받거나 다른 사람에게 옮겼을때 Material이 될것같습니다.

Reply:
개러지에 아이템을 담을 때에는 거래 가능한 아이템을 거래 불가능한 아이템으로 바꾸는 것이 기능이어서 그래요. TradableMaterial을 그대로 올리면 FungibleItemGarage.Item이 TradableMaterial이 되는 가능성이 열리게 되는데요. 이러면 하역할 때 거래 가능한 아이템으로 하역돼요.
//
When loading items into the garage, it's a feature to turn a tradable item into a non-tradable item. If you load a tradable material as it is, it opens up the possibility that FungibleItemGarage.Item becomes a tradable material, which means that when you unload it, it will be unloaded as a tradable item.